plugins/common/tls/utils.go

```diff
@@ -36,3 +43,58 @@ func ParseTLSVersion(version string) (uint16, error) {
 	sort.Strings(available)
 	return 0, fmt.Errorf("unsupported version %q (available: %s)", version, strings.Join(available, ","))
 }
+
+func readFile(filename string) []byte {
+	octets, err := os.ReadFile(filename)
+	if err != nil {
+		panic(fmt.Sprintf("reading %q: %v", filename, err))
+	}
+	return octets
+}
+
+func ReadCertificate(filename string) string {
+	octets := readFile(filename)
```

Comment:
AIUI, this will end up feeding both the public and private keys to the first argument of `tls.X509KeyPair()` or multiple blocks if more than one certificate and/or private key is in the combined file. https://pkg.go.dev/crypto/tls#X509KeyPair suggests that this should only be a single cert block. I suggest not relying on implementation dependent behavior in `tls.X509KeyPair()` and make this code only return one cert block.